azbus/receiver.go

```diff
@@ -302,12 +348,86 @@ func (r *Receiver) ReceiveMessages(handler Handler) error {
 
 }
 
+func (r *Receiver) receiveMessagesInParallel() error {
+
+	if len(r.handlers) != r.Cfg.NumberOfReceivedMessages {
+		return fmt.Errorf("%s: Number of Handlers %d is not equal to %d", r, len(r.handlers), r.Cfg.NumberOfReceivedMessages)
+	}
+
+	err := r.Open()
+	if err != nil {
+		azerr := fmt.Errorf("%s: ReceiveMessage failure: %w", r, NewAzbusError(err))
+		r.log.Infof("%s", azerr)
+		return azerr
+	}
+	r.log.Debugf(
+		"NumberOfReceivedMessages %d, RenewMessageLock: %v",
+		r.Cfg.NumberOfReceivedMessages,
+		r.Cfg.RenewMessageLock,
+	)
+
+	// Start all the workers
+	msgs := make(chan *ReceivedMessage, r.Cfg.NumberOfReceivedMessages)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	var wg sync.WaitGroup
+	for i := 0; i < r.Cfg.NumberOfReceivedMessages; i++ {
+		go func(rctx context.Context, ii int, rr *Receiver) {
```

Comment:
minor: 

I know its a contentious subject, but i feel strongly about anonymous functions.

This indentation is quite large already, 

1. func 
2. for
3. anonymous function
4. for
5. select
6. case
7. if

That is too many levels of indentation.

Can we move the anonymous function into its own function?

Putting as minor, as it is not overly hard to read.